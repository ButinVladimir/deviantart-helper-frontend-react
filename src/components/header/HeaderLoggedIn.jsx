import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import Navbar from 'react-bulma-components/lib/components/navbar';
import Button from 'react-bulma-components/lib/components/button';
import * as routes from '../../consts/routes';

export default function HeaderLoggedIn({
  deviationsLoading,
  revokeLoading,
  userName,
  userIcon,
  toggleMenuHandler,
  deviationsLoadHandler,
  revokeHandler,
}) {
  return (
    <Fragment>
      <Navbar.Brand>
        <Navbar.Item>
          <img src={userIcon} alt={userName} />
        </Navbar.Item>
        <Navbar.Burger onClick={toggleMenuHandler} />
      </Navbar.Brand>

      <Navbar.Menu>
        <Navbar.Container>
          <Navbar.Item dropdown hoverable>
            <Navbar.Link>Application</Navbar.Link>
            <Navbar.Dropdown>
              <Navbar.Item renderAs={() => <NavLink className="navbar-item" to={routes.USER_INFO}>User info</NavLink>} />
            </Navbar.Dropdown>
          </Navbar.Item>

          <Navbar.Item dropdown hoverable>
            <Navbar.Link>Deviations</Navbar.Link>
            <Navbar.Dropdown>
              <Navbar.Item renderAs={() => <NavLink className="navbar-item" to={routes.DEVIATIONS_BROWSE}>Browse</NavLink>} />
              <Navbar.Item renderAs={() => <NavLink className="navbar-item" to={routes.DEVIATIONS_STATISTICS}>Statistics</NavLink>} />
            </Navbar.Dropdown>
          </Navbar.Item>
        </Navbar.Container>

        <Navbar.Container position="end">
          <Navbar.Item>
            <Button.Group>
              <Button color="light" loading={deviationsLoading} onClick={deviationsLoadHandler}>Refresh data</Button>
              <Button color="danger" loading={revokeLoading} onClick={revokeHandler}>Log out</Button>
            </Button.Group>
          </Navbar.Item>
        </Navbar.Container>
      </Navbar.Menu>
    </Fragment>
  );
}

HeaderLoggedIn.propTypes = {
  deviationsLoading: PropTypes.bool.isRequired,
  revokeLoading: PropTypes.bool.isRequired,
  userName: PropTypes.string,
  userIcon: PropTypes.string,
  toggleMenuHandler: PropTypes.func.isRequired,
  deviationsLoadHandler: PropTypes.func.isRequired,
  revokeHandler: PropTypes.func.isRequired,
};

HeaderLoggedIn.defaultProps = {
  userName: '',
  userIcon: '',
};
