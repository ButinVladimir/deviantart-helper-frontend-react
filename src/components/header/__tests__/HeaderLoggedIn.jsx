import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';
import HeaderLoggedIn from '../HeaderLoggedIn';

describe('HeaderLoggedIn', () => {
  it('can be rendered correctly', () => {
    const wrapper = mount(
      <MemoryRouter>
        <HeaderLoggedIn
          startFetchDataLoading={false}
          revokeLoading={false}
          userName="User Name"
          userIcon="/path/to/user/icon"
          toggleMenuHandler={() => {}}
          startFetchDataHandler={() => {}}
          revokeHandler={() => {}}
        />
      </MemoryRouter>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('can handle toggling menu', () => {
    const toggleMenuHandler = jest.fn();
    const wrapper = shallow(
      <HeaderLoggedIn
        startFetchDataLoading={false}
        revokeLoading={false}
        userName="User Name"
        userIcon="/path/to/user/icon"
        toggleMenuHandler={toggleMenuHandler}
        startFetchDataHandler={() => {}}
        revokeHandler={() => {}}
      />,
    );

    wrapper.find('#toggle-menu-button').simulate('click');

    expect(toggleMenuHandler).toHaveBeenCalled();
  });

  it('can handle revoking session', () => {
    const revokeHandler = jest.fn();
    const wrapper = shallow(
      <HeaderLoggedIn
        startFetchDataLoading={false}
        revokeLoading={false}
        userName="User Name"
        userIcon="/path/to/user/icon"
        toggleMenuHandler={() => {}}
        startFetchDataHandler={() => {}}
        revokeHandler={revokeHandler}
      />,
    );

    wrapper.find('#revoke-button').simulate('click');

    expect(revokeHandler).toHaveBeenCalled();
  });

  it('can handle fetching data', () => {
    const startFetchDataHandler = jest.fn();
    const wrapper = shallow(
      <HeaderLoggedIn
        startFetchDataLoading={false}
        revokeLoading={false}
        userName="User Name"
        userIcon="/path/to/user/icon"
        toggleMenuHandler={() => {}}
        startFetchDataHandler={startFetchDataHandler}
        revokeHandler={() => {}}
      />,
    );

    wrapper.find('#start-fetch-data-button').simulate('click');

    expect(startFetchDataHandler).toHaveBeenCalled();
  });
});
