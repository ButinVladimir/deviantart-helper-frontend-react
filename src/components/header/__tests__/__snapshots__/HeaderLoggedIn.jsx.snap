// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLoggedIn can be rendered correctly 1`] = `
Array [
  <div
    className="navbar-brand"
    style={Object {}}
  >
    <a
      className="navbar-item"
      style={Object {}}
    >
      <img
        alt="User Name"
        src="/path/to/user/icon"
      />
    </a>
    <div
      className="navbar-burger is-active"
      id="toggle-menu-button"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "outline": "none",
        }
      }
      tabIndex="0"
    >
      <span />
      <span />
      <span />
    </div>
  </div>,
  <div
    className="navbar-menu is-active"
    style={Object {}}
  >
    <div
      className="navbar-start"
      style={Object {}}
    >
      <span
        className="navbar-item has-dropdown is-hoverable"
        style={Object {}}
      >
        <span
          className="navbar-link"
          style={Object {}}
        >
          Application
        </span>
        <span
          className="navbar-dropdown"
          style={Object {}}
        >
          <a
            aria-current={null}
            className="navbar-item"
            href="/user/info"
            onClick={[Function]}
          >
            User info
          </a>
        </span>
      </span>
      <span
        className="navbar-item has-dropdown is-hoverable"
        style={Object {}}
      >
        <span
          className="navbar-link"
          style={Object {}}
        >
          Deviations
        </span>
        <span
          className="navbar-dropdown"
          style={Object {}}
        >
          <a
            aria-current={null}
            className="navbar-item"
            href="/deviations/browse"
            onClick={[Function]}
          >
            Browse
          </a>
          <a
            aria-current={null}
            className="navbar-item"
            href="/deviations/statistics"
            onClick={[Function]}
          >
            Detailed statistics
          </a>
          <a
            aria-current={null}
            className="navbar-item"
            href="/deviations/total"
            onClick={[Function]}
          >
            Total statistics
          </a>
        </span>
      </span>
    </div>
    <div
      className="navbar-end"
      style={Object {}}
    >
      <a
        className="navbar-item"
        style={Object {}}
      >
        <div
          className="buttons"
        >
          <button
            className="is-light button"
            disabled={false}
            id="start-fetch-data-button"
            onClick={[Function]}
            style={Object {}}
            tabIndex={0}
          >
            Request data fetching
          </button>
          <button
            className="is-danger button"
            disabled={false}
            id="revoke-button"
            onClick={[Function]}
            style={Object {}}
            tabIndex={0}
          >
            Log out
          </button>
        </div>
      </a>
    </div>
  </div>,
]
`;
